types/parse/index.d.ts

```diff
@@ -79,6 +80,11 @@ declare enum ErrorCode {
 
 declare namespace Parse {
 
+    // Helper types for extracting an extended generic type from an
+    // (Parse.)Object or Array.
+    type Unarray<T> = T extends Array<infer U> ? U : T;
```

Comment:
Similarly.
```suggestion
    type Unarray<T> = T extends Array<infer U> ? U : never;
```

Reply:
I agree, except that I'm (mis)using the side-effect of a non-unarrayable type to fallback to it's given type by default. 

I've modified Unarray and Unobject to fallback to `never` by default, unless provided otherwise.